app/code/community/Bolt/Boltpay/Controller/Traits/WebHookTrait.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Bolt magento plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Trait Bolt_Boltpay_Controller_Traits_WebHookTrait
+ *
+ * Defines generalized actions associated with web hooks
+ *
+ * @method Mage_Core_Controller_Response_Http getResponse()
+ * @method Mage_Core_Model_Layout getLayout()
+ */
+trait Bolt_Boltpay_Controller_Traits_WebHookTrait {
+    use Bolt_Boltpay_BoltGlobalTrait;
+
+    /**
+     * @var string The body of the request made to this controller
```

Comment:
you expect json as input, why not json_decode here?

Reply:
@daisy1754 I will take it as a sign that attempting to implement this nitpick is leading to spectacular  unit test failures:
https://circleci.com/gh/BoltApp/bolt-magento1/2799?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
https://circleci.com/gh/BoltApp/bolt-magento1/2803?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Moreover, we want the payload to be an honest representation of what the Bolt server sent in the request body.  Bolt is sending a json string, not a decoded PHP object.

However, for convenience, I will provide a utility public accessor method.

Reply:
> you expect json as input, why not json_decode here?

@daisy1754 I will take it as a sign that attempting to implement this nitpick is leading to spectacular  unit test failures:
https://circleci.com/gh/BoltApp/bolt-magento1/2799?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
https://circleci.com/gh/BoltApp/bolt-magento1/2803?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Moreover, we want the payload to be an honest representation of what the Bolt server sent in the request body.  Bolt is sending a json string, not a decoded PHP object.

However, for convenience, I have provided a utility public accessor method.